### PR TITLE
Feature/sequence number

### PR DESCRIPTION
**Pull request by Tigraine**

### CODE REVIEW

File: BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class BitmovinAnalytics {
      */
     public func detachPlayer() {
         eventDataDispatcher.disable()
-        stateMachine.reset()
+        stateMachine = StateMachine(config: self.config)
```
Comment (zachmanc) [inline BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift]: what is the thought behind this change? With it, we seem to be leaking StateMachine objects. Steps to reproduce are 

1) Using the sample app, start playback using either AVPlayer or Bitmovin Player. Set a break point in StateMachine.deinit() method
2) Press the reload button.
3) Notice that our break point in deinit is never called
4) Play the new stream
5) Pause the stream and wait for the next heartbeat event to occur. 
6) Notice that two heartbeat events are sent.

File: BitmovinAnalyticsCollector/Classes/data/EventData.swift [hunk 2]
```diff
@@ -55,6 +55,8 @@ public class EventData: Codable {
     var pageLoadType: Int = 1
     var pageLoadTime: Int64 = 0
     var version: String?
+    var sequenceNumber: Int32 = 0
+    var platform: String = "iOS"
```
Comment (zachmanc) [inline BitmovinAnalyticsCollector/Classes/data/EventData.swift]: I assume we want to support tvOS here as well? If so, you can use this macro to detect 

```swift
#if os(iOS)
// compiles for iOS
#elseif os(tvOS)
// compiles for tvOS
#endif
```

**Comment by Tigraine** [comment]
Thanks Cory.
The first change was made to avoid having to clean stuff up like that. But
it seems that was not enough.

**Comment by zachmanc** [comment]
Sounds good, I think the timer might be what is causing the object to stay around, but I would need to look at it a bit closer. It might be safer to just reset the state machine like we are doing before and address this in a future PR

File: BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift (hunk 1 above)
Comment (Tigraine) [reply BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift]: I reverted the change for now.

File: BitmovinAnalyticsCollector/Classes/data/EventData.swift (hunk 2 above)
Comment (Tigraine) [reply BitmovinAnalyticsCollector/Classes/data/EventData.swift]: Thanks implemented :)

File: BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift (hunk 1 above)
Comment (zachmanc) [reply BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift]: Do you need to reset the sequenceNumber to 0 when the reset occurs? Other than that, it looks good!

**Pull request merged**